kgraphql/src/main/kotlin/com/apurebase/kgraphql/schema/execution/ArgumentTransformer.kt

```diff
@@ -42,6 +42,15 @@ open class ArgumentTransformer(val schema : DefaultSchema) {
                     params.getValue(valueField.name.value) to transformValue(paramType, valueField.value, variables)
                 }.toMap()
 
+                val valueMapNames = valueMap.keys.map { it.name }
+                val missingNonOptionalInputs = params
+                        .filter { (key, value) -> !value.isOptional && !valueMapNames.contains(key) }
+
+                require(missingNonOptionalInputs.isEmpty()) {
+                    val inputs = missingNonOptionalInputs.keys.joinToString(",")
+                    throw GraphQLError("You are missing non optional input fields: $inputs")
+                }
```

Comment:
Just to save some loops. This could be changed to:

```kotlin
val missingNonOptionalInputs = params.values
    .filter { !it.isOptional && !valueMap.containsKey(it) }

if (missingNonOptionalInputs.isNotEmpty()) {
    val inputs = missingNonOptionalInputs.map { it.name }.joinToString(",")
    throw GraphQLError("You are missing non optional input fields: $inputs", value)
}
```

And when throwing an `GraphQLError` we always want to include the `ASTNode` when available.

After these changes this looks great :+1: